plugins/modules/zos_operator.py

```diff
@@ -247,26 +239,13 @@ def run_module():
         # but it could still be a bad/invalid command.
         # As long as there are more than 2 lines, it's worth looking through.
         if int(result["rc"]) == 0:
-            if len(short_str) > 2:
+            if len(result["content"]) > 2:
                 result["changed"] = True
-                for linetocheck in short_str:
-                    if "invalid" in linetocheck.lower():
-                        result["exception"] = "Invalid detected: " + linetocheck
-                        result["changed"] = False
-                        module.fail_json(msg=result["exception"], **result)
-                    elif "error" in linetocheck.lower():
-                        result["exception"] = "Error detected: " + linetocheck
-                        result["changed"] = False
-                        module.fail_json(msg=result["exception"], **result)
-                    elif "unidentifiable" in linetocheck.lower():
-                        result["exception"] = "Unidentifiable detected: " + linetocheck
-                        result["changed"] = False
-                        module.fail_json(msg=result["exception"], **result)
             else:
                 module.fail_json(msg="Expected response to be more than 2 lines.", **result)
         else:
             module.fail_json(
-                msg="Non-zero response received: " + str(result["rc"]), **result
+                msg="Non-zero response received: " + str(result["rc"] + "\n" + ". Stderr: " + str(error)), **result
```

Comment:
My second comment is why are you adding "Stderr" like so, as if its a key:value pair we want to pass back. You have the dictionary **result from what I can see has ALMOST everything you need. 

What I would do is either populate the `**result` further such that

Or create a dynamic one with what the docs state will returned. For example:
```
module.fail_json(msg=(
                        "A non-zero return code was received : {0}. Review the response for more details.").format(result["rc"]),
                    stderr=str(error),
                    stderr_lines=str(error).splitlines(),
                   cmd=result["cmd"],
                   stdout=str(stdout),
                   stdout_lines=str(stdout).splitlines(),
                   wait_time_s...
                   elapsed_time....
                  and so on....)
```
My personal preference is you start with result dictionary and keep appending as you go along, start by putting all the information you know is needed per our [response contract](https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/modules/zos_operator.html#return-values). 

And the nice thing is if you build it as you move along in the code you can be append to the dictionary when values are not None. For example, in your `module.fail_json` you did not check if this `Stderr: " + str(error))` was not none. 

If you build the dictionary as you go along you can use the checks you have that are checking if stdout or stderr are not none and then at the very end in the exception you can just return `module.fail_json(msg="Your mess here", **result)`

Reply:
Here is my response two both comments. I change to the result dictionary to the [response contract](https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/modules/zos_operator.html#return-values). Just change the content for stderr and stderr lines, also verify if are None just put all the content.

Reply:
I run the pipeline for [Ansible Certification] (<img width="1245" alt="Captura de pantalla 2023-05-05 a la(s) 10 33 08" src="https://user-images.githubusercontent.com/68956970/236515503-764163de-3953-4ed4-9a86-da7fb86c44ae.png">) and is ready.

Reply:
@AndreMarcel99 looks great thanks so much for the added effort !!!